src/artwork.c

```diff
@@ -392,6 +393,64 @@ static struct online_source musicbrainz_source =
   };
 
 
+/* -------------------------------- HELPERS -------------------------------- */
+
+#define TMP_BUF_SIZE 1024
+static int artwork_read_byurl(struct evbuffer *evbuf, const char *url);
+
+/* Reads an artwork url tags from the json responce within evbuffer, then if found 
+ * makes call to read image from url and replace evbuffer contents with image data
+ *
+ * @out evbuf     Json data passed | replaced with Image data returned
+ * @in  url       URL for the image
+ * @return        ART_FMT_* on success, ART_E_ERROR otherwise
+ */
+
+static int
+artwork_read_json_byurl(struct evbuffer *evbuf, const char *url)
```

Comment:
Rename, you are not reading artwork. Maybe streamurl_json_parse(). I also suggest reducing the function scope to only parse. Then let the caller get the artwork.

Reply:
Good point, that may also remove the forward declaration above.  Just didn't want calling function to get too complicated.